src/Fixer/ClassNotation/StaticPrivateMethodFixer.php

```diff
@@ -0,0 +1,288 @@
+    }
+
+    private function baz()
+    {
+        return 1;
+    }
+}
+'
+                ),
+            ],
+            null,
+            'Risky when method contains dynamic generated calls to the instance, or the method is dynamically referenced.'
+        );
+    }
+
+    public function getPriority()
+    {
+        // Must run after ProtectedToPrivateFixer, before StaticLambdaFixer
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_PRIVATE]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        $end = \count($tokens) - 3; // min. number of tokens to form a class candidate to fix
+        for ($index = $end; $index > 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_CLASS)) {
+                continue;
+            }
+
+            $classOpen = $tokens->getNextTokenOfKind($index, ['{']);
+            $classClose = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classOpen);
+
+            $this->fixClass($tokens, $tokensAnalyzer, $classOpen, $classClose);
+        }
+    }
+
+    /**
+     * @param Tokens         $tokens
+     * @param TokensAnalyzer $tokensAnalyzer
+     * @param int            $classOpen
+     * @param int            $classClose
+     */
+    private function fixClass(Tokens $tokens, TokensAnalyzer $tokensAnalyzer, $classOpen, $classClose)
+    {
+        $fixedMethods = [];
+        for ($index = $classClose - 1; $index > $classOpen + 1; --$index) {
+            if (!$tokens[$index]->equals('}')) {
+                continue;
+            }
+
+            $methodClose = $index;
+            $methodOpen = $tokens->findBlockStart(Tokens::BLOCK_TYPE_CURLY_BRACE, $methodClose);
+            $functionKeywordIndex = $tokens->getPrevTokenOfKind($methodOpen, [[T_FUNCTION]]);
+            $methodNameIndex = $tokens->getNextMeaningfulToken($functionKeywordIndex);
+            $methodName = $tokens[$methodNameIndex]->getContent();
+
+            $index = $methodOpen;
+
+            $prevTokenIndex = $tokens->getPrevMeaningfulToken($functionKeywordIndex);
+            $prevPrevTokenIndex = $tokens->getPrevMeaningfulToken($prevTokenIndex);
+            if ($tokens[$prevTokenIndex]->isGivenKind(T_ABSTRACT) || $tokens[$prevPrevTokenIndex]->isGivenKind(T_ABSTRACT)) {
+                continue;
+            }
+
+            if ($this->skipMethod($tokens, $functionKeywordIndex, $methodOpen, $methodClose)) {
+                continue;
+            }
+
+            $fixedMethods[$methodName] = true;
+
+            $tokens->insertAt($functionKeywordIndex, [new Token([T_STATIC, 'static']), new Token([T_WHITESPACE, ' '])]);
```

Comment:
How does this work with final methods?

Even though it doesn't do much, private methods *can* be final.

Reply:
Fixed, thanks